code/game/machinery/harvester.dm

```diff
@@ -0,0 +1,160 @@
+/obj/machinery/harvester
+	name = "organ harvester"
+	desc = "An advanced machine used for harvesting organs and limbs from the deceased."
+	density = TRUE
+	icon = 'icons/obj/machines/harvester.dmi'
+	icon_state = "harvester"
+	verb_say = "states"
+	state_open = FALSE
+	idle_power_usage = 50
+	circuit = /obj/item/circuitboard/machine/harvester
+	light_color = LIGHT_COLOR_BLUE
+	var/interval = 20
+	var/harvesting = FALSE
+	var/list/operation_order = list() //Order of wich we harvest limbs.
+	var/allow_clothing = FALSE
+	var/allow_living = FALSE
+
+/obj/machinery/harvester/Initialize()
+	. = ..()
+
+/obj/machinery/harvester/RefreshParts()
+	interval = 0
+	var/max_time = 40
+	for(var/obj/item/stock_parts/micro_laser/L in component_parts)
+		max_time -= L.rating
+	if(max_time <= 0)
+		max_time = 1 //ungodly, but whatever
+	interval = max_time
+
+/obj/machinery/harvester/update_icon(warming_up)
+	if(warming_up)
+		icon_state = initial(icon_state)+"-charging"
+		return
+	if(state_open)
+		icon_state = initial(icon_state)+"-open"
+	else if(harvesting)
+		icon_state = initial(icon_state)+"-active"
+	else
+		icon_state = initial(icon_state)
+
+/obj/machinery/harvester/open_machine(drop = TRUE)
+	. = ..()
+	harvesting = FALSE
+
+/obj/machinery/harvester/attack_hand(mob/user)
+	if(state_open)
+		close_machine()
+	else if(!harvesting)
+		open_machine()
+
+/obj/machinery/harvester/AltClick(mob/user)
+	if(harvesting || !user || !isliving(user) || state_open)
+		return
+	if(can_harvest())
+		start_harvest()
+
+/obj/machinery/harvester/proc/can_harvest()
+	if(!powered(EQUIP) || state_open || !occupant || !iscarbon(occupant))
+		return
+	var/mob/living/carbon/C = occupant
+	if(!allow_clothing)
+		for(var/obj/item/I in C.held_items + C.get_equipped_items())
+			if(!(I.item_flags & NODROP))
+				say("Subject may not have abiotic items on.")
+				playsound(src, 'sound/machines/buzz-sigh.ogg', 30, 1)
+				return
+	if(ishuman(C))
+		var/mob/living/carbon/human/H = C
+		if(H.dna && H.dna.species && !(MOB_ORGANIC in H.dna.species.inherent_biotypes))
+			say("Subject is not organic.")
+			playsound(src, 'sound/machines/buzz-sigh.ogg', 30, 1)
+			return
```

Comment:
check the mob_biotypes in the mob, not the species; the species one simply adds/removes it to the mob